### PR TITLE
DOC Update Documentation in "missing_data" 54028

### LINKED ISSUE #54028: DOC:  Add `NAType` and `NaTType` references in docs for `pandas.api.typing`
In #53934 I suggested we move `NAType` and `NaTType` into `pandas.api.typing`, which was done in #53958 .

But the PR did not include a docs update.

I think it would be worthwhile to add a sentence or two to https://pandas.pydata.org/docs/dev/user_guide/missing_data.html to describe these new public types and their purpose.

### PR DESCRIPTION
**Pull request by omar-elbaz**
Closes #54028 DOC: Add NAType and NaTType references in docs for pandas.api.typing
Adds reference notes for importing NaType and NaTType

### CODE REVIEW

File: doc/source/user_guide/missing_data.rst [hunk 1]
```diff
@@ -91,6 +91,10 @@ See :ref:`integer_na` for more.
 
 Datetimes
 ---------
+.. note::
+   If you are adding type checking to your application, you may need access to ``NaTType`` and ``NAType``.
+
+   ``NaTTYPE`` and ``NAType`` have now been added and can be imported from "pandas.api.typing".
```
Comment (jreback) [inline doc/source/user_guide/missing_data.rst:97]: you don't need this sentence here

Comment (omar-elbaz) [reply doc/source/user_guide/missing_data.rst:97]: > you don't need this sentence here

Which sentence are you referring to?

Comment (omar-elbaz) [reply doc/source/user_guide/missing_data.rst:97]: > you don't need this sentence here
If you mean the entire comment would you suggest somewhere else where it fits better?

**Review by jreback — APPROVED** [review]
lgtm

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @omar-elbaz

**Pull request merged**